src/app/k-tools/k-tool-form-dialog/k-tool-form-dialog.component.ts

```diff
@@ -0,0 +1,47 @@
+import { Component, Inject } from '@angular/core';
```

Comment:
Why have a component just to handle dialog functions? Cant k-tool-form have these functions?

Reply:
For re-usability. I'd like to be able to slap the k tool form somewhere else on the site in the future if need be. 

And also to answer the why of there being a whole component for the dialog thing - there are the two injections that happen - MatDialogRef and MAT_DIALOG_DATA that only work when the component is created via the Material dialog service (k-tools.component.ts line 76)

Reply:
Sounds good for now - I think we can move this to a shared component though instead of doing this in every form?

Reply:
yeah there is a lot of duplicate boilerplate sort of code... but I don't know how we would generalize some of that stuff. You mean KToolFormDialogComponent?

Reply:
Yeah, I'm not sure either but I think its worth looking into if/when we run out of stories.